server/subscribe.go

```diff
@@ -538,21 +574,39 @@ func httpChannelDeleteSubscription(p *Plugin, w http.ResponseWriter, r *http.Req
 		return http.StatusBadRequest, errors.Wrap(err, "bad subscription id")
 	}
 
-	if err := p.hasPermissionToManageSubscription(mattermostUserId, subscription.ChannelId); err != nil {
+	if err = p.hasPermissionToManageSubscription(mattermostUserId, subscription.ChannelId); err != nil {
 		return http.StatusForbidden, errors.Wrap(err, "you don't have permission to manage subscriptions")
 	}
 
 	if _, err := p.API.GetChannelMember(subscription.ChannelId, mattermostUserId); err != nil {
 		return http.StatusForbidden, errors.New("Not a member of the channel specified")
 	}
 
-	if err := p.removeChannelSubscription(subscriptionId); err != nil {
+	if err = p.removeChannelSubscription(subscriptionId); err != nil {
```

Comment:
The variable shadowing issues I've come across like this situation are usually a symptom caused by `error` vs `model.AppError`. While we're touching this code, for errors returned from `p.API` methods, 1/5 we should rename the error `appErr` and favor the
```
err = p.some_func()
if err != nil {

appErr = p.API.some_api_func()
if appErr != nil {
```
pattern instead of the one-lined
```
if err = some_func(); err != nil {
```
for each if statement in these three functions.

Reply:
Yes, that's what I forgot.  Will make another commit soon.  Thanks

Reply:
Are you suggesting we change all inline if statements or only those that return an appErr?

Reply:
All of the inline if statements